src/java/picard/analysis/CollectAlignmentSummaryMetrics.java

```diff
@@ -116,6 +116,10 @@ public static void main(final String[] argv) {
                     "in the file are aligned then alignment summary metrics collection will fail.");
         }
 
+        if (super.REFERENCE_SEQUENCE != null && this.REFERENCE_SEQUENCE == null) {
```

Comment:
Ah ok. We might want to add some logic in the command line parser to handle an overriden option that is being set to null. Of course that is assuming you would never want to override and option and set it to null.  In any case this fix seems good to me for this particular case.